hal/src/rtl872x/radio_common.cpp

```diff
@@ -122,4 +122,18 @@ int selectRadioAntenna(radio_antenna_type antenna) {
     return SYSTEM_ERROR_NONE;
 }
 
+int getRadioAntenna(radio_antenna_type * antenna) {
+    if (!antenna) {
+        return SYSTEM_ERROR_INVALID_ARGUMENT;
+    }
+    uint8_t dctAntenna = 0xff;
+    CHECK(dct_read_app_data_copy(DCT_RADIO_ANTENNA_OFFSET, &dctAntenna, sizeof(dctAntenna)));
```

Comment:
I'd do exactly the same thing as currently done in `initRadioAntenna()` and return default normally unless dct read value is exactly INTERNAL or EXTERNAL and replace that block with getRadioAntenna() so that we don't have duplicate logic.